src/EnrollCallAndSmsController.js

```diff
@@ -274,8 +282,24 @@ function (Okta, FormController, Footer, PhoneTextBox, TextBox, CountryUtil, Form
         noButtonBar: true,
         autoSave: true,
         className: getClassName(factorType),
+
+        showWarning: function (msg) {
+          clearTimeout(warningTimeout);
+          this.clearWarnings();
+          this.add(warningTemplate, '.o-form-error-container', {options: {warning: msg}});
+        },
+        clearWarnings: function () {
+          this.$('.okta-form-infobox-warning').remove();
+        },
+
         initialize: function () {
+          this.listenTo(this.model, 'change:ableToResend', function (model, ableToResend) {
+            if (ableToResend) {
+              this.showWarning(warning);
+            }
```

Comment:
I think we need the `else` to clear the warning right? Otherwise what happens if you click Re-send or Redial? Is the warning cleared?

Reply:
this is handled by clearErrors events, which clear both warning and errors once you clicked re-send/re-dial, I think it is necessary to put it there since we have scenarios of both warning and errors happen in the same time